src/contracts/Protocol/modifiers/TokenBar.sol

```diff
@@ -0,0 +1,28 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Interfaces
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "../interfaces/IClaimController.sol";
+
+abstract contract TokenBar is Ownable {
+    address public metricToken;
+
+    /// @notice Throw if user tries to claim with balance lower than threshold
+    error NotEnoughTokens();
+
+    function setMetricToken(address _metricToken) public onlyOwner {
```

Comment:
WE have this same function in 3 different places now - the Vault and costController and now here - can we combine these somehow?

Reply:
Yeah will look into that, single source of truth for metric address seems like a better approach